include/exadg/incompressible_navier_stokes/spatial_discretization/operator_coupled.cpp

```diff
@@ -1208,24 +1208,30 @@ OperatorCoupled<dim, Number>::apply_inverse_negative_laplace_operator(VectorType
   {
     // solve a linear system of equations for negative Laplace operator to given (relative)
     // tolerance using the PCG method
-    VectorType const * pointer_to_src = &src;
+    dst = 0.0;
     if(this->is_pressure_level_undefined())
     {
-      VectorType vector_zero_mean;
-      vector_zero_mean = src;
-
+      VectorType vector_zero_mean = src;
       if(laplace_operator->operator_is_singular())
       {
         dealii::VectorTools::subtract_mean_value(vector_zero_mean);
       }
-
-      pointer_to_src = &vector_zero_mean;
+      else
+      {
+        AssertThrow(false,
+                    dealii::ExcMessage(
+                      "Undefined pressure level always leads to singular Laplace operator."));
```

Comment:
Note that this `else`-branch + `AssertThrow()` is new. I think that `this->is_pressure_level_undefined()` implies `laplace_operator->operator_is_singular()`. I would like to add the assert.

Reply:
I think the additional assert is irritating, because `is_pressure_level_is_undefined()` is used when setting up the operator data for the laplace operator. I think the code should look like this:


```C++
  else
  {
     VectorType const * pointer_to_src = &src;

     // manipulate src-vector if operator is singular
      VectorType vector_zero_mean;
      if(laplace_operator->operator_is_singular())
      {
        vector_zero_mean = src;
        dealii::VectorTools::subtract_mean_value(vector_zero_mean);
        pointer_to_src = &vector_zero_mean;
      }

      // call the solver
     ...
   }
```

Does that make sense?

Reply:
Ehm so I think if you replace `else` in your code block with `if(this->is_pressure_level_undefined())`, it is what we had above. But lets stay closer to the version we had originally. I will push a suggestion asap! I think I know what you mean :)

Reply:
The `else` in the pseudo-code above comes from the outer if/else branch of the function `apply_negative_laplace_operator()`.

So I really want to avoid the usage of `is_pressure_level_undefined()` and `operator_is_singular()`, where I agree that this is irritating.